src/components/utils.js

```diff
@@ -0,0 +1,26 @@
+import { filter, flatten, values } from 'lodash';
+
+export function takenThePledge(record) {
+  return record.pledged ? ' has taken the pledge.' : ' has not taken the pledge.';
+}
+
+export const totalPledgedInState = itemsInState => Object.keys(itemsInState)
```

Comment:
As long as we're using redux, why not setup a reducer to do these?

Reply:
I'm using these in two places. For getting the totals for the state cards: it's just deriving something from the data already in state. The other place is for making the popovers on the map... I could store what is currently hovered/focused in the state and then use a selector to grab this data, but since this is such a fleeting state change and wholly a concern of the map, it didnt make sense to me to lift it up.